tests/network/sriov.go

```diff
@@ -719,3 +721,40 @@ func checkDefaultInterfaceInPod(vmi *v1.VirtualMachineInstance) error {
 
 	return nil
 }
+
+// createSRIOVVmiOnNode creates a VMI on the specified node, connected to the specified SR-IOV network.
+func createSRIOVVmiOnNode(nodeName, networkName, cidr string) (*v1.VirtualMachineInstance, error) {
+	// Explicitly choose different random mac addresses instead of relying on kubemacpool to do it:
+	// 1) we don't at the moment deploy kubemacpool in kind providers
+	// 2) even if we would do, it's probably a good idea to have the suite not depend on this fact
+	//
+	// This step is needed to guarantee that no VFs on the PF carry a duplicate MAC address that may affect
+	// ability of VMIs to send and receive ICMP packets on their ports.
+	mac, err := GenerateRandomMac()
+	if err != nil {
+		return nil, err
+	}
+
+	// manually configure IP/link on sriov interfaces because there is
+	// no DHCP server to serve the address to the guest
+	vmi := newSRIOVVmi([]string{networkName}, cloudInitNetworkDataWithStaticIPsByMac(networkName, mac.String(), cidr))
+	const secondaryInterfaceIndex = 1
+	vmi.Spec.Domain.Devices.Interfaces[secondaryInterfaceIndex].MacAddress = mac.String()
+
+	vmi = tests.CreateVmiOnNode(vmi, nodeName)
+
+	return vmi, nil
+}
+
+func sriovNodeName(sriovResourceName string) (string, error) {
```

Comment:
I would change this function to return a list of nodes and pick one of them at the caller,
and rename it to `listNodeByResource`.
It may help the reader understand where VMs are created.

Reply:
The function abstracts how the node is selected. The reader at the caller side should either be satisfied with understanding that a suitable node is returned or, if he insist, can go into the function implementation and examine how it is done. Exposing the list reduces the abstraction and requires the caller to do some logic to chose one.

I think the existing proposal is better.